Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

#Answer

To calculate the Pearson correlation coefficient between two variables, such as the amount of time students spend studying for an exam and their final exam scores, you can use the following formula:


r = n(∑xy) − (∑x) (∑y) /(squareroot[n∑x** - (∑x) ** ][n∑y ** -(∑y) ** ]


Where:


n is the number of data points (students in this case).

∑xy is the sum of the products of each student's study time and exam score.

∑x is the sum of all the study times.

∑y is the sum of all the exam scores.

∑x**  is the sum of the squares of the study times.

∑y** is the sum of the squares of the exam scores.


Here's how you can interpret the result:

* If r is close to 1: This indicates a strong positive linear relationship between study time and exam scores. In other words, as study time increases, exam scores tend to increase.

* If r is close to -1: This indicates a strong negative linear relationship between study time and exam scores. In other words, as study time increases, exam scores tend to decrease.

* If r is close to 0: This suggests a weak or no linear relationship between study time and exam scores. There may be little to no correlation between the two variables.

* The sign of r (positive or negative) indicates the direction of the relationship. A positive sign means that as one variable increases, the other tends to increase, and a negative sign means that as one variable increases, the other tends to decrease.

* The magnitude of r (how close it is to 1 or -1) indicates the strength of the relationship. The closer r is to 1 or -1, the stronger the linear relationship.

So, after calculating r for your specific data, you can interpret the correlation coefficient to understand the nature and strength of the relationship between study time and exam scores among the students in your study.



                      -------------------------------------------------------------------

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [7]:
import scipy.stats

# Sample data for sleep and job satisfaction
sleep = [6, 7, 8, 5, 6, 7, 8, 5, 4, 6]
job_satisfaction = [7, 8, 9, 5, 6, 8, 9, 4, 3, 7]

# Calculate Spearman's rank correlation
spearman_corr, _ = scipy.stats.spearmanr(sleep, job_satisfaction)

print("Spearman's rank correlation coefficient:", spearman_corr)

Spearman's rank correlation coefficient: 0.9875771574795098


Here ρ is close to 1: This indicates a strong positive monotonic relationship between sleep and job satisfaction. In other words, as the amount of sleep increases, job satisfaction tends to increase in a monotonic fashion. This suggests that people who get more sleep tend to report higher job satisfaction.

                      -------------------------------------------------------------------

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

In [28]:
#Answer

import numpy as np
from scipy import stats

# assuming  Sample data for exercise hours and BMI
exercise_hours = [2, 3, 4, 3, 5, 2, 1, 4, 3, 5, 2, 6, 3, 4, 2, 5, 1, 3, 6, 4, 5, 2, 1, 4, 3, 5, 2, 6, 3, 4, 2, 5, 1, 3, 6, 4, 5, 2, 1, 4, 3, 5, 2, 6, 3, 4, 5, 5,6,2]
bmi = [28, 24, 22, 26, 30, 29, 31, 25, 27, 23, 28, 21, 26, 24, 29, 22, 30, 26, 20, 22, 21, 27, 29, 23, 26, 20, 25, 21, 26, 28, 24, 20, 27, 22, 29, 23, 26, 30, 21, 24, 22, 28, 31, 25, 27, 29, 30,30,32,22]

# Calculate Pearson correlation coefficient
pearson_corr, _ = stats.pearsonr(exercise_hours, bmi)

# Calculate Spearman's rank correlation
spearman_corr, _ = stats.spearmanr(exercise_hours, bmi)

print("Pearson correlation coefficient:", pearson_corr)
print("Spearman's rank correlation coefficient:", spearman_corr)


Pearson correlation coefficient: -0.2943991578993425
Spearman's rank correlation coefficient: -0.3022013280675571



Interpreting the results:

Pearson Correlation Coefficient:

The Pearson correlation coefficient is approximately -0.294.
* It is negative, indicating a weak negative linear relationship between the number of hours of exercise per week and body mass index (BMI). This means that, on average, as the number of hours of exercise per week increases, BMI tends to decrease slightly.
* The magnitude of -0.294 suggests a relatively weak linear association, which means that the linear correlation between these two variables is not very strong.

Spearman's Rank Correlation Coefficient:

* The Spearman's rank correlation coefficient is approximately -0.302.
* It is also negative, indicating a weak negative monotonic relationship between exercise hours and BMI. This suggests that, on average, as exercise hours increase, BMI tends to decrease in a monotonic fashion, which aligns with the Pearson correlation results.
The magnitude of -0.302 suggests a relatively weak monotonic association, which means that there is a weak tendency for BMI to decrease as exercise hours increase in a monotonic manner.

                      -------------------------------------------------------------------

Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [27]:
#Answer

import scipy.stats

# lets assume Sample data for TV hours and physical activity
tv_hours = [2, 3, 4, 3, 5, 2, 1, 4, 3, 5, 2,4,1, 3, 6, 3, 4, 2, 5, 1, 3, 6, 4, 5, 2, 1, 4, 3, 5, 2, 6, 3, 4, 2, 5, 1, 3, 6, 4, 5, 2, 1, 4, 3, 5, 2, 6, 3, 4, 5]
physical_activity = [3, 2, 1, 3, 1, 4, 5,4, 1,2,1, 2, 3, 1, 4, 2, 3, 2, 4, 1, 5, 3, 2, 1, 4, 4, 5, 3, 2, 1, 4, 2, 1, 3, 2, 3, 4, 5, 1, 3, 4, 2, 2, 1, 3, 5, 4, 3, 2, 1]

# Calculate Pearson correlation coefficient
pearson_corr, _ = scipy.stats.pearsonr(tv_hours, physical_activity)

print("Pearson correlation coefficient:", pearson_corr)


Pearson correlation coefficient: -0.009745574363240125


Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:

  Age(Years)                 Soft drink Preference
   25                          Coke
   42                          Pepsi
   37                          Mountain dew
   19                          Coke
   31                          Pepsi
   28                          Coke


In [ ]:
........................there is no question given, hence leaving the answer as blank....................................

                       -------------------------------------------------------------------

Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables

In [26]:
#Answer

import scipy.stats

# let's Sample data for sales calls per day and sales per week
sales_calls_per_day = [20, 15, 25, 30, 18, 22, 27, 19, 21, 26, 24, 23, 17, 28, 16, 29, 30, 22, 18, 25, 19, 20, 17, 23, 21, 26, 16, 27, 24, 28]
sales_per_week = [5, 3, 6, 7, 4, 6, 8, 5, 6, 7, 7, 6, 4, 8, 3, 8, 9, 6, 4, 7, 5, 5, 4, 6, 6, 7, 3, 8, 6, 7]

# Calculate Pearson correlation coefficient
pearson_corr, _ = scipy.stats.pearsonr(sales_calls_per_day, sales_per_week)

print("Pearson correlation coefficient:", pearson_corr)


Pearson correlation coefficient: 0.9523971601611595


                        -------------------------------------------------------------------